homeassistant/components/teslemetry/config_flow.py

```diff
@@ -27,41 +27,78 @@
+            access_token=user_input[CONF_ACCESS_TOKEN],
+        )
+        try:
+            await teslemetry.test()
+        except InvalidToken:
+            return {CONF_ACCESS_TOKEN: "invalid_access_token"}
+        except SubscriptionRequired:
+            return {"base": "subscription_required"}
+        except ClientConnectionError:
+            return {"base": "cannot_connect"}
+        except TeslaFleetError as e:
+            LOGGER.error(str(e))
+            return {"base": "unknown"}
+        return {}
 
     async def async_step_user(
         self, user_input: Mapping[str, Any] | None = None
-    ) -> config_entries.ConfigFlowResult:
+    ) -> ConfigFlowResult:
         """Get configuration from the user."""
         errors: dict[str, str] = {}
-        if user_input:
-            teslemetry = Teslemetry(
-                session=async_get_clientsession(self.hass),
-                access_token=user_input[CONF_ACCESS_TOKEN],
+        if user_input and not (errors := await self.async_auth(user_input)):
+            self._abort_if_unique_id_configured()
+            return self.async_create_entry(
+                title="Teslemetry",
+                data=user_input,
             )
-            try:
-                await teslemetry.test()
-            except InvalidToken:
-                errors[CONF_ACCESS_TOKEN] = "invalid_access_token"
-            except SubscriptionRequired:
-                errors["base"] = "subscription_required"
-            except ClientConnectionError:
-                errors["base"] = "cannot_connect"
-            except TeslaFleetError as e:
-                LOGGER.exception(str(e))
-                errors["base"] = "unknown"
-            else:
-                return self.async_create_entry(
-                    title="Teslemetry",
-                    data=user_input,
-                )
 
         return self.async_show_form(
             step_id="user",
             data_schema=TESLEMETRY_SCHEMA,
             description_placeholders=DESCRIPTION_PLACEHOLDERS,
             errors=errors,
         )
+
+    async def async_step_reauth(
+        self, entry_data: Mapping[str, Any]
+    ) -> ConfigFlowResult:
+        """Handle reauth on failure."""
+        self._entry = self.hass.config_entries.async_get_entry(self.context["entry_id"])
+        return await self.async_step_reauth_confirm()
+
+    async def async_step_reauth_confirm(
+        self, user_input: Mapping[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle users reauth credentials."""
+
+        assert self._entry
+        errors: dict[str, str] | None = None
+
+        if user_input and not (errors := await self.async_auth(user_input)):
+            self.hass.config_entries.async_update_entry(
+                self._entry,
+                data=user_input,
+            )
+            self.hass.async_create_task(
+                self.hass.config_entries.async_reload(self._entry.entry_id)
+            )
```

Comment:
```suggestion
                self.hass.config_entries.async_reload(self._entry.entry_id)
```
Is `async_create_task()` required?

Reply:
I am unsure if its required, but this is how its done in adobe, airvisual, apple_tv, cloudflare, and so many others. Testing to see what difference it makes now.

Reply:
This has been removed.